torch_geometric/datasets/qm9_featurized.py

```diff
@@ -0,0 +1,329 @@
+        return None
+
+    @property
+    def raw_file_names(self) -> List[str]:
+        try:
+            import rdkit  # noqa
+            return ['gdb9.sdf', 'gdb9.sdf.csv', 'uncharacterized.txt']
+        except ImportError:
+            return ['qm9_v3_featurized.pt']
+
+    @property
+    def processed_file_names(self) -> str:
+        return 'data_v3_featurized.pt'
+
+    def download(self):
+        try:
+            import rdkit  # noqa
+            file_path = download_url(self.raw_url, self.raw_dir)
+            extract_zip(file_path, self.raw_dir)
+            os.unlink(file_path)
+
+            file_path = download_url(self.raw_url2, self.raw_dir)
+            os.rename(osp.join(self.raw_dir, '3195404'),
+                      osp.join(self.raw_dir, 'uncharacterized.txt'))
+        except ImportError:
+            path = download_url(self.processed_url, self.raw_dir)
+            extract_zip(path, self.raw_dir)
+            os.unlink(path)
+
+    def process(self):
+        try:
+            import rdkit
+            from rdkit import Chem, RDLogger
+            from rdkit.Chem.rdchem import BondType as BT
+            from rdkit.Chem.rdchem import HybridizationType
+            RDLogger.DisableLog('rdApp.*')
+
+        except ImportError:
+            rdkit = None
+
+        if rdkit is None:
+            print(("Using a pre-processed version of the dataset. Please "
+                   "install 'rdkit' to alternatively process the raw data."),
+                  file=sys.stderr)
+
+            data_list = torch.load(self.raw_paths[0])
+            data_list = [Data(**data_dict) for data_dict in data_list]
+
+            if self.pre_filter is not None:
+                data_list = [d for d in data_list if self.pre_filter(d)]
+
+            if self.pre_transform is not None:
+                data_list = [self.pre_transform(d) for d in data_list]
+
+            self.save(data_list, self.processed_paths[0])
+            return
+
+        types = {'H': 0, 'C': 1, 'N': 2, 'O': 3, 'F': 4}
+        bonds = {BT.SINGLE: 0, BT.DOUBLE: 1, BT.TRIPLE: 2, BT.AROMATIC: 3}
+
+        with open(self.raw_paths[1], 'r') as f:
+            target = f.read().split('\n')[1:-1]
+            target = [[float(x) for x in line.split(',')[1:20]]
+                      for line in target]
+            target = torch.tensor(target, dtype=torch.float)
+            target = torch.cat([target[:, 3:], target[:, :3]], dim=-1)
+            target = target * conversion.view(1, -1)
+
+        with open(self.raw_paths[2], 'r') as f:
+            skip = [int(x.split()[0]) - 1 for x in f.read().split('\n')[9:-2]]
+
+        suppl = Chem.SDMolSupplier(self.raw_paths[0], removeHs=False,
+                                   sanitize=True)
+
+        data_list = []
+        for i, mol in enumerate(tqdm(suppl)):
+            if i in skip or mol == None:
+                continue
+
+            mol = Chem.AddHs(mol)
+            N = mol.GetNumAtoms()
```

Comment:
Hi, just a little suggestion, how about we set `sanitize` and `Chem.AddHs` by parameters? Like keeping `sanitize=False` and `add_hs=False` in default, and `sanitize=True` and `add_hs=True` when you want to make it featurized.

```python
suppl = Chem.SDMolSupplier(self.raw_paths[0], removeHs=False, sanitize=self.sanitize)

data_list = []
for i, mol in enumerate(tqdm(suppl)):
    if i in skip or mol == None:
        continue
    if self.add_hs:
        mol = Chem.AddHs(mol)
    N = mol.GetNumAtoms()
```

Reply:
Thanks for the suggestion, that's definitely cleaner than having two separate classes. Given that both AddHs and sanitization are necessary to have a chemically correct dataset, I think it would be easier to just include a single `featurize` parameter that would control both of these. This also avoids needing to upload four separate pre-processed versions of the dataset for users without RDKit installed. As before, it will still be necessary to upload a zip file containing a pre-processed version of the featurized dataset to https://data.pyg.org/datasets/qm9_v3_featurized.zip. I've updated the PR accordingly, and please let me know if you have any other suggestions.

Reply:
I've finished up the clarification changes, and all tests are passing. I ended up changing the URL for the pre-processed dataset to 'https://data.pyg.org/datasets/qm9_v3_ft.zip' for flake8 compliance, so this is where someone with access should upload the processed featurized dataset. Please let me know if you have any other suggestions, and otherwise I think this should be ready to merge.

Reply:
@rusty1s is there anything else I should do regarding this PR? I've been periodically updating it to match changes in the main repo (mostly changelog updates) but haven't made any other changes since the PR was approved by @xnuohz

Reply:
Please don't bother with updating the PR, I don't think there will be any merge conflicts besides `CHANGELOG` and we can fix them before merging. I'll try to get this PR in this week.